packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -145,10 +271,33 @@ export class Trigger {
         };
         return { tags };
       });
+      const pullRequest: CfnPipeline.GitPullRequestFilterProperty[] | undefined = pullRequestFilter?.map(filter => {
+        const branches: CfnPipeline.GitBranchFilterCriteriaProperty | undefined =
+          filter.branchesExcludes?.length || filter.branchesIncludes?.length ? {
+            // set to undefined if empty array because CloudFormation does not accept empty array
+            excludes: filter.branchesExcludes?.length ? filter.branchesExcludes : undefined,
+            includes: filter.branchesIncludes?.length ? filter.branchesIncludes : undefined,
+          } : undefined;
+        const filePaths: CfnPipeline.GitFilePathFilterCriteriaProperty | undefined =
+          filter.filePathsExcludes?.length || filter.filePathsIncludes?.length ? {
+            // set to undefined if empty array because CloudFormation does not accept empty array
+            excludes: filter.filePathsExcludes?.length ? filter.filePathsExcludes : undefined,
+            includes: filter.filePathsIncludes?.length ? filter.filePathsIncludes : undefined,
+          } : undefined;
+        // set to all events if empty array or undefined because CloudFormation does not accept empty array and undefined
+        const events: string[] = filter.events?.length ? Array.from(new Set(filter.events)) : [
+          GitPullRequestEvent.OPEN,
+          GitPullRequestEvent.UPDATED,
+          GitPullRequestEvent.CLOSED,
+        ];
```

Comment:
The CDK code and CFn with events as an empty array or undefined will occur.

```
The trigger configuration for the source action is not valid. Make sure to choose one trigger configuration for each source action.
```

In addition, `You must select at least one pull request event.` is displayed in AWS console.